compiler/src/test/scala/org/scalajs/nscplugin/test/StaticForwardersASTTest.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.nscplugin.test
+
+import org.scalajs.nscplugin.test.util._
+
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.junit.Test
+
+// scalastyle:off line.size.limit
+
+class StaticForwardersASTTest extends JSASTTest  {
+
+  @Test
+  def emitStaticForwardersInExistingClass(): Unit = {
+    val BarName = MethodName("bar", List(IntRef), IntRef)
+    val BarDefaultAccessorName = MethodName("bar$default$1", Nil, IntRef)
+
+    val CtorDefaultAccessorName = MethodName("$lessinit$greater$default$1", Nil, IntRef)
+
+    val FoobarName = MethodName("foobar", List(IntRef), IntRef)
+    val FoobarDefaultAccessorName = MethodName("foobar$default$1", Nil, IntRef)
+
+    """
+    import scala.scalajs.js, js.annotation._
+
+    class Foo(val y: Int = 10)
+
+    object Foo {
+      def bar(x: Int = 5): Int = x + 1
+
+      @js.native
+      @JSGlobal("foobar")
+      def foobar(x: Int = 5): Int = js.native
+    }
+    """.hasExactly(1, "static accessor for bar") {
+      case MethodDef(flags, MethodIdent(BarName), _, _, _, _) if flags.namespace.isStatic =>
+    }.hasExactly(1, "static accessor for the default accessor of bar") {
+      case MethodDef(flags, MethodIdent(BarDefaultAccessorName), _, _, _, _) if flags.namespace.isStatic =>
+    }.hasExactly(1, "static accessor for the default accessor of the constructor") {
+      case MethodDef(flags, MethodIdent(CtorDefaultAccessorName), _, _, _, _) if flags.namespace.isStatic =>
+    }.hasNot("static accessor for foobar") {
+      case MethodDef(flags, MethodIdent(FoobarName), _, _, _, _) if flags.namespace.isStatic =>
```

Comment:
`FoobarName` is never "tested" to be correct. In the sense that there is high risk that its name could be a typo. As an alternative; have you considered a test that collects all the method names and namespaces and asserts the resulting set/map?

IIUC that should not be too brittle (i.e break on unrelated changes) but much more robust against typos.

Reply:
Good point. I updated the test using your suggestion.